.github/workflows/test-docker-deployments.yml

```diff
@@ -0,0 +1,111 @@
+name: Test Deployments
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+# cancel older, redundant runs of same workflow on same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  docker-deployments:
+    name: Test the Docker Deployments
+    runs-on: ubuntu-latest
+    env:
+      ZIGGY_UID: 1001  # let container EUID run-as GHA "runner" user to share cache, etc.
+      ZITI_PWD: ziggylicious
+      ZITI_CTRL_ADVERTISED_ADDRESS: ctrl1.127.21.71.0.sslip.io
+      ZITI_ROUTER_ADVERTISED_ADDRESS: router1.127.21.71.0.sslip.io
+      ZITI_ROUTER_NAME: router1
+      # define the compose project files: compose.yml and compose.test.yml files for ziti-controller and ziti-router
+      COMPOSE_FILE: ${{ github.workspace }}/dist/docker-images/ziti-controller/compose.yml:${{ github.workspace }}/dist/docker-images/ziti-controller/compose.test.yml:${{ github.workspace }}/dist/docker-images/ziti-router/compose.yml:${{ github.workspace }}/dist/docker-images/ziti-router/compose.test.yml
+      ZITI_CONTROLLER_IMAGE: ziti-controller:local
+      ZITI_ROUTER_IMAGE: ziti-router:local
+    steps:
+      - name: Shallow checkout
```

Comment:
I strongly dislike tests that can't be run easily and succinctly locally. though it looks nice within the GitHub UI, these sorts of things should be easily performed locally.

I'd much rather see most of these steps condensed into a local script that someone can run locally.

Reply:
I'm also hungry for a way to run Actions locally. I'd guess it would frequently save me a significant amount of time. It's probably possible for most of our jobs. I'd have to go back to the drawing board on this one because so much of it is scaffolding from other jobs that, like this one, are intended to run in GitHub. Adapting all of our jobs to shell scripts doesn't seem like the right solution unless the job doesn't benefit from any GitHub Actions.